### PR TITLE
Add types for query, remove type assertions and switch to inference where possible

### PR DESCRIPTION
**Pull request by calirvine**
Connects https://github.com/benvinegar/counterscale/issues/13

Sorry about the formatting changes 🤔  I think they're just prettier defaults. I'll try to revert them all to the previous style (unless you want to add a prettier config, then I can just apply that to the branch). 

### Notes on changes
- I introduced an invariant fn for asserting on types returned from the cloudflare API
  - Made invariant checks a bit optimistic, as I thought looping over the whole dataset might get expensive 🤷‍♂️ 
  - Invariants throw, but accessing malformed data will probably also throw later, so I figure that's ok.
- Based types on [documentation](https://developers.cloudflare.com/analytics/analytics-engine/sql-api/#table-structure), haven't tested them.

Hopefully changes aren't controversial, but let me know.

### CODE REVIEW

**Comment by calirvine** [comment]
@benvinegar These types should work, but the prettier changes make this un-reviewable 😅  will revert them tonight before marking this ready for review. These types can probably be more ergonomic than this, but it's better than `any`.

File: app/routes/dashboard.test.tsx [hunk 1]
```diff
@@ -133,14 +133,14 @@ describe("Dashboard route", () => {
                 countByDevice: [['Desktop', 3]],
                 viewsGroupedByInterval: [
                     ['2024-01-11 00:00:00', 4],
-                    ['2024-01-26 00:00:00', 0],
```
Comment (calirvine) [inline app/routes/dashboard.test.tsx:136]: Flakey test, reported in https://github.com/benvinegar/counterscale/issues/27

**Comment by benvinegar** [comment]
I spent a solid 30 minutes going over this, mostly because my TypeScript experience is pretty basic, and I had to read up on a bunch of concepts (like user-defined type guards).

I think this hardens the API-calling surface a ton, so +1 from me.

(Also should probably just put some Prettier rules in here already and add it to the pre-commit stuff.)

**Comment by benvinegar** [comment]
Also – thanks! You've made the first external contribution, which I'm super glad for, because this project isn't going anywhere with just me 😬

**Comment by calirvine** [comment]
Invariants are nice, they're basically the same as `as X` but they add a runtime check. The one I included (and which I usually use) is a paired back version of [tiny-invariant](https://github.com/alexreardon/tiny-invariant).

**Comment by benvinegar** [comment]
FYI I had to revert this PR in 6c7108 because #30. I didn't do any deep debugging but I agree with the reporters that the `invariant` code seems likely, and we should probably have more tests to cover these range of cases.

Let me know if that's something you wanna poke at – I'm happy to, but I also want to make room for folks.

**Comment by calirvine** [comment]
First contributor, first to break everything. Casual 😅 . Agreed it's the invariants, though they should be throwing very clear errors. I wonder where those are getting swallowed 🤔 .

I see which invariant is broken, it's `getVisitorCountByColumn`. 

I was working last night on improving that method to infer the return type based on the name of the column passed in, I'll update that branch to be a bit less aggressive in enforcing type correctness.

**Pull request merged**